javascript/javascript3/week3/shoppingCartBest.js

```diff
@@ -0,0 +1,55 @@
+class Product {
+    constructor(name, price) {
+      this.name = name;
+      this.price = price;
+    }
+  }
+  
+  class ShoppingCart {
+    constructor() {
+      this.products = [];
+    }
+  
+    addProduct(product) {
+      // Implement functionality here
+      this.products.push(product);
+    }
+  
+    removeProduct(product) {
+      const filtered = this.products.filter(value => {
+        return value !== product;
+      });
```

Comment:
You need to assign this new product list to `this.products`, otherwise, it's not doing anything.
`filtered` is also not a good name.